### PR TITLE
[Improvement] use incubating version in docs#5114

### PR DESCRIPTION
**Pull request by lsyulong**
### What changes were proposed in this pull request?

[Improvement] use incubating version in docs#5114

### Why are the changes needed?

Gravitino has become an Apache Incubator project since version 0.6.0, so it is necessary to follow the Apache versioning convention like 0.6.0-incubating instead of 0.6.0.

Fix: #5114 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

No need

### CODE REVIEW

File: clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java [hunk 1]
```diff
@@ -84,6 +84,24 @@ private GravitinoClient(
...
+   * Constructs a new GravitinoClient with the given URI, authenticator and AuthDataProvider.
+   *
+   * @param uri The base URI for the Gravitino API.
+   * @param authDataProvider The provider of the data which is used for authentication.
+   * @param checkVersion Whether to check the version of the Gravitino server. Gravitino does not
+   *     support the case that the client-side version is higher than the server-side version.
+   * @param headers The base header for Gravitino API.
+   * @throws NoSuchMetalakeException if the metalake with specified name does not exist.
+   */
+  private GravitinoClient(
```
Comment (yuqi1129) [inline clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java:97]: Are these expected changes in this PR?

Comment (lsyulong) [reply clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java:97]: No, anything other than the document can be ignored

Comment (lsyulong) [reply clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java:97]: Do I need to close PR, fork the repository again, make modifications based on this repository, and submit PR again

Comment (yuqi1129) [reply clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java:97]: @lsyulong 
You can remove this unrelated code directly and do not need to close this PR.

**Pull request closed without merging**